src/super_gradients/compliance/check_compliance.py

```diff
@@ -0,0 +1,122 @@
+"""
+A Python script that check's the compliance of Hosts (OS and Hardware) for Deci's infrastructure.
+"""
+
+import os
+import sys
+from pip._internal.operations.freeze import freeze
+from logging import getLogger, DEBUG
+from typing import List, Dict
+from pathlib import Path
+from packaging.version import Version
+
+logger = getLogger('deci-compliance')
+logger.setLevel(DEBUG)
+
+
+def verify_os() -> List[str]:
+    """Verifying operating system name and platform"""
+    name = os.name
+    platform = sys.platform
+    logger.info(f'OS Name: {name}')
+    logger.info(f'OS Platform: {platform}')
+    if 'linux' not in platform.lower():
+        return ['Deci officially supports only Linux kernels. Some features may not work as expected.']
+    return []
+
+
+def get_libs_requirements() -> List[str]:
+    """Read requirement.txt from the root, and split it as a list of libs/version"""
+    with open(Path(__file__).parent.parent.parent.parent / "requirements.txt", "r") as f:
+        requirements_str = f.read()
+    return requirements_str.split("\n")
```

Comment:
why not use f.readlines() instead? this implementation may fail if someone uses other type of line breaker